utils/generic-components.stanza

```diff
@@ -652,25 +651,27 @@ public pcb-module micro-usb-connector :
   schematic-group(self) = micro-usb
 
 ; USB2 device connection via USBC connector
-public pcb-module usb2-on-a-usb-c-connector :
-  port usb-2 : usb-2
-  public inst con : ocdb/belfuse/SS-52400-003/component
-
-  net VBUS (usb-2.vbus.vdd con.vbus)
-  net (usb-2.vbus.gnd con.GND)
-
-  net D- (usb-2.data.N con.DN1, con.DN2)
-  net D+ (usb-2.data.P con.DP1, con.DP2)
+public pcb-enum ocdb/generic-components/usb-c-connector:
+  Belfuse-SS-52400-003
+  korean-hroparts-elec-TYPE-C-31-M-12
 
-  ; RC filter connection for chassis gnd
-  res-strap(con.SHIELD, con.GND, 100.0e3)
-  cap-strap(con.SHIELD, con.GND, 0.1e-6)
-
-  ; 5V Downstream USBC device configuration
-  res-strap(con.CC1, con.GND, 5.1e3)
-  res-strap(con.CC2, con.GND, 5.1e3)
-
-  property(con.vbus) = PowerSupplyPin(5.0, 0.5)
+public defn  usb2-on-a-usb-c-connector ():
+  usb2-on-a-usb-c-connector(Belfuse-SS-52400-003)
+public pcb-module usb2-on-a-usb-c-connector (conn-mpn:ocdb/generic-components/usb-c-connector):
```

Comment:
I like the idea of providing the connector as an argument to this module. 

You can avoid the enum + switch, by directly providing the instantiable:
```
 public pcb-module usb2-on-a-usb-c-connector (con-inst:Instantiable) :
  inst con : con-inst
```
Called with 
```
usb2-on-a-usb-c-connector(ocdb/belfuse/SS-52400-003/module)
```

I would drop the `usb2` definitions in the component defpackages, and use this module to map `usb2` by instantiating the connector and using the configuration function.